assessment-api/assessment-actors/src/main/scala/org/sunbird/actors/QuestionSetActor.scala

```diff
@@ -161,9 +163,50 @@ class QuestionSetActor @Inject()(implicit oec: OntologyEngineContext) extends Ba
 			DataNode.systemUpdate(request, response,"questionSet", Some(HierarchyManager.getHierarchy))
 		}).map(node => ResponseHandler.OK.put("identifier", identifier).put("status", "success"))
 	}
-
 	def copy(request: Request): Future[Response] ={
 		RequestUtil.restrictProperties(request)
 		CopyManager.copy(request)
 	}
+	def updateComment(request: Request): Future[Response] = {
+		AssessmentManager.getValidatedNodeForUpdateComment(request, "ERR_QUESTION_SET_UPDATE_COMMENT").flatMap(_ => updateInnerComment(request))
+	}
+	def updateInnerComment(request: Request): Future[Response] = {
+		val readReq = new Request(request)
+		val writeReq = new Request(request)
+
+		readReq.getRequest.remove("comments")
+		writeReq.getRequest.remove("comments")
+
+		val comments = request.getRequest.get("comments").asInstanceOf[java.util.ArrayList[java.util.Map[String, Object]]].asScala.toList
+		val futureResults = Future.traverse(comments) { comment =>
+			val identifier = comment.get("identifier").asInstanceOf[String]
+			val rejectComment = comment.get("comment")
+
+			readReq.getRequest.clear()
+			readReq.getRequest.put("rejectComment", rejectComment)
+			readReq.getContext.replace("identifier", identifier)
+
+			RequestUtil.validateRequest(readReq)
```

Comment:
why readReq need to be validated?

Reply:
Piyush said he did it because he saw it being done elsewhere. But I understand that since this is an internal request which is purely generated by actor code, this validation isn't required. Will be removed.